### DISCUSSION TITLE
Can I use `winget configure` with locally-defined DSC?

### DISCUSSION THREAD

**Discussion by cthlo** [microsoft/winget-cli#4230] [Q&A]
### Scenario
There is some custom installation logic for our application. We'd like to implement that logic in DSC (instead of scripting it in [PSDscResources/Script](https://learn.microsoft.com/en-us/powershell/dsc/reference/resources/windows/scriptresource?view=dsc-1.1&viewFallbackFrom=dsc-2.0)).

Can I author our `.dsc.yaml` to utilize the local DSC?
Thanks!

**Comment by Trenly** [comment]
If you have built and installed your DSC resource, you should be able to reference the module directly. If I recall correctly, winget will try to download and install any resources not currently present on the machine, but it should find them if they are already installed

**Reply by cthlo** [reply]
I see.
I was hoping to minimize the set-up before `winget configure`, especially if we want to use Dev Home to clone and set up via the configuration file.
But thanks for the advice!

**Comment by stephengillie** [comment]


**Reply by Trenly** [reply]
@stephengillie - I think that @cthlo was asking about a DSC Resource, not necessarily the configuration specification. For example - the `Microsoft.WinGet.DSC/WinGetPackage` is a publicly available DSC Resource in the `Microsoft.WinGet.DSC` module. There is also a `PSDscResources/Script` DSC Resource which @cthlo referenced in the initial post (and which is used in the winget-pkgs .configurations file). 

For reference -
* DSC Module - A group of DSC Resources in a portable format with metadata on how the resources are meant to be used
* DSC Resource - A PowerShell module which contains methods for getting, setting, and testing desired state
  * https://learn.microsoft.com/en-us/powershell/dsc/resources/resources?view=dsc-2.0
* DSC Configuration - This is the file used for specifying the desired state, either as the YAML which WinGet interprets, or as the raw PowerShell
  * https://learn.microsoft.com/en-us/powershell/dsc/configurations/configurations?view=dsc-2.0#configuration-syntax
